src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2281,37 +2113,30 @@ MavlinkReceiver::handle_message_collision(mavlink_message_t *msg)
 	t.altitude_minimum_delta = collision.altitude_minimum_delta;
 	t.horizontal_minimum_delta = collision.horizontal_minimum_delta;
 
-	if (_collision_report_pub == nullptr) {
-		_collision_report_pub = orb_advertise(ORB_ID(collision_report), &t);
-
-	} else {
-		orb_publish(ORB_ID(collision_report), _collision_report_pub, &t);
-	}
+	_collision_report_pub.publish(t);
 }
 
 void
 MavlinkReceiver::handle_message_gps_rtcm_data(mavlink_message_t *msg)
 {
-	mavlink_gps_rtcm_data_t gps_rtcm_data_msg = {};
+	mavlink_gps_rtcm_data_t gps_rtcm_data_msg;
```

Comment:
We lost the initializer on this one.

Reply:
We don't have them when the immediate next call is the appropriate mavlink decode which then unconditionally sets every single field.